src/ray/raylet/event/events.h

```diff
@@ -0,0 +1,302 @@
+#ifndef RAY_EVENT_EVENTS_H
+#define RAY_EVENT_EVENTS_H
+
+#include <memory>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+#include "ray/id.h"
+
+namespace ray {
+
+namespace events {
+
+using SubscriptionID = UniqueID;
+
+/// The enum of event status.
+enum class EventStatus { pending, cancelled, finished };
+
+struct EventEdge;
+template <typename TEvent>
+class EventsManager;
+
+class IEventNode {
+ public:
+  /// Finish an input edge of the node.
+  ///
+  /// \param in_edge The input edge.
+  /// \return Void.
+  virtual void FinishInEdge(std::shared_ptr<EventEdge> &in_edge) = 0;
```

Comment:
I don't understand all of this graph terminology. What is an "edge"? Does the node manager really need anything more complex than a mapping from raylet client to a set of ObjectIDs that that client is subscribed to?